keys/keys.go

```diff
@@ -0,0 +1,149 @@
+package keys
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Shopify/sarama"
+	"github.com/google/uuid"
+)
+
+type (
+	// Key represents a kafka key.
+	Key interface {
+		sarama.Encoder
+	}
+
+	// CreateTimer is a create timer or delete timer event, value should be a messaging.CreateTimer or nil.
+	CreateTimer struct {
+		domain    string
+		timerUUID uuid.UUID
+	}
+
+	// ExecuteTimer is an execute timer event, value should be a common.ExecuteTimer or nil.
+	ExecuteTimer struct {
+		domain    string
+		timerUUID uuid.UUID
+	}
+
+	// Dummy is a dummy event so a node knows when its caught up after a repartition. Value should be nil.
+	Dummy struct{}
+)
+
+const (
+	createTimerLabel  = "create"
+	createTimerFields = 2
+
+	executeTimerLabel  = "execute"
+	executeTimerFields = 2
+
+	dummyLabel  = "dummy"
+	dummyFields = 0
+)
+
+// NewCreateTimer creates a new CreateTimer key.
+func NewCreateTimer(domain string, timerUUID uuid.UUID) Key {
+	return CreateTimer{domain: domain, timerUUID: timerUUID}
+}
+
+// Encode returns an encoded timer.
+func (c CreateTimer) Encode() ([]byte, error) {
+	return []byte(fmt.Sprintf("%v:%v:%v", createTimerLabel, c.domain, c.timerUUID)), nil
```

Comment:
Consider using byteBuilder here its faster under the covers then Sprintf and since this is called on every single publish you want it to be really fast

Reply:
Ok. I'll also consider just using protos.